src/PackageManagement/NuGetPackageManager.cs

```diff
@@ -382,6 +385,7 @@ private void InitializePackagesFolderInfo(string packagesFolderPath)
                     var action = NuGetProjectAction.CreateInstallProjectAction(updateToIdentity, primarySources.FirstOrDefault());
 
                     var lowLevelActions = new List<NuGetProjectAction>();
+                    lowLevelActions.Add(NuGetProjectAction.CreateUninstallProjectAction(installedPackageReference.PackageIdentity));
```

Comment:
What if the version to be installed is the same version of the already installed version? I don't see logic to distinguish update-package Id, update-package -reinstall. Looks like it's forcely uninstall first anyways. 

Is this by design?